src/descriptor/pegin/legacy_pegin.rs

```diff
@@ -143,14 +143,21 @@ impl<Pk: MiniscriptKey> LegacyPegin<Pk> {
         timelock: bitcoin::Sequence,
         desc: Descriptor<Pk, CovenantExt<CovExtArgs>>,
     ) -> Self {
-        let fed_ms = BtcMiniscript::from_ast(BtcTerminal::Multi(fed_k, fed_pks.clone()))
-            .expect("Multi type check can't fail");
+        let fed_ms = BtcMiniscript::from_ast(BtcTerminal::Multi(
+            bitcoin_miniscript::Threshold::new(fed_k, fed_pks.clone()).expect("TODO"),
```

Comment:
we should probably accept Threshold and RelLockTime as parameter instead of returning `Result<Self,Something>`

Reply:
Definitely agreed. I think this `.expect("TODO")` was probably intended to signal this change.

Reply:
Done in https://github.com/ElementsProject/elements-miniscript/pull/90/commits/9d6be01d71be1c8168e5309f78cc0fd76175c53a